soundata/datasets/tau2020uas_mobile.py

```diff
@@ -0,0 +1,961 @@
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="tau2020uas_mobile",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.cached_property
+    def _metadata(self):
+
+        metadata_path = os.path.join(
+            self.data_home,
+            "TAU-urban-acoustic-scenes-2020-mobile-development",
+            "meta.csv",
+        )
+
+        splits = [
+            "development.train",
+            "development.evaluate",
+            "evaluation",
+        ]
+
+        metadata_index = {}
+
+        with open(metadata_path) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter="\t")
+            next(csv_reader)
+            for row in csv_reader:
+                file_name = os.path.basename(row[0])
+                clip_id = os.path.basename(file_name).replace(".wav", "")
+                scene_label = row[1]
+                identifier = row[2]
+                source_label = row[3]
+                city = identifier.split("-")[0]
+                metadata_index[clip_id] = {
+                    "scene_label": scene_label,
+                    "city": city,
+                    "identifier": identifier,
+                    "source_label": source_label,
+                }
+
+        for split in splits:
+            subset = split.split(".")[0]
+            evaluation_setup_path = (
+                "TAU-urban-acoustic-scenes-2020-mobile-{}/evaluation_setup".format(
+                    subset
+                )
+            )
+            if subset == "development":
+                fold = split.split(".")[1]
+                evaluation_setup_file = os.path.join(
+                    self.data_home,
+                    evaluation_setup_path,
+                    "fold1_{}.csv".format(fold),
+                )
+            else:
+                evaluation_setup_file = os.path.join(
+                    self.data_home, evaluation_setup_path, "fold1_test.csv"
+                )
+
+            with open(evaluation_setup_file) as csv_file:
+                csv_reader = csv.reader(csv_file, delimiter="\t")
+                next(csv_reader)
+                for row in csv_reader:
+                    file_name = os.path.basename(row[0])
+                    clip_id = os.path.basename(file_name).replace(".wav", "")
+
+                    if subset != "development":
+                        metadata_index[clip_id] = {
+                            "scene_label": None,
+                            "city": None,
+                            "identifier": None,
+                            "source_label": None,
+                        }
+
+                    metadata_index[clip_id]["split"] = split
+
+        return metadata_index
```

Comment:
Just to see if I follow: 1) you have a `metadata` file with all `clips` from the development subset, which you read first and fill each `field` of the metadata, but that file doesn't have the split information; 2) you get into the subsets of the `development` set and fill the split information for those, and add the `clip` ids of the evaluation?

Reply:
Yes, but note that the step 2) includes splits from both development and evaluation datasets.